ff-macros/src/lib.rs

```diff
@@ -0,0 +1,88 @@
+#![deny(
+    warnings,
+    unused,
+    future_incompatible,
+    nonstandard_style,
+    rust_2018_idioms
+)]
+#![forbid(unsafe_code)]
+#![recursion_limit = "128"]
```

Comment:
Could a comment be added for why we set a recursion limit here?
IIRC, you had mentioned that its something to do with constraints on compile time macros at the moment

Reply:
I'll have to check, but I think it's not an issue in ff-macros

Reply:
Addressed